### PR TITLE
[IMP] account: The new draft invoice has the sale order as the source document

### PR DESCRIPTION
**Pull request by rap-odoo**
Description of the issue/feature this PR addresses:
- Source document on invoice create through a credit note with the modify option

<strong>Task: </strong>https://www.odoo.com/web#id=23075&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
<strong>Pad: </strong>https://pad.odoo.com/p/r.97a8b834d41a9ec102117cd46d8f2813

Current behavior before PR:
- The new draft invoice doesn't have a source document and can't be reached from the SO

Desired behavior after PR is merged:
- The new draft invoice has the sale order as the source document and reachable from the sale order.
## 

I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

```
 -now the new draft invoice is reachable from the sale order.
```

### CODE REVIEW

File: addons/account/models/account_invoice.py [hunk 1]
```diff
@@ -872,6 +872,8 @@ def _refund_cleanup_lines(self, lines):
                     values[name] = line[name]
                 elif name == 'invoice_line_tax_ids':
                     values[name] = [(6, 0, line[name].ids)]
+                elif name == 'sale_line_ids':
```
Comment (tde-banana-odoo) [inline addons/account/models/account_invoice.py]: Why this change ?

Comment (gpa-odoo) [inline addons/account/models/account_invoice.py]: This change done for when refund invoice with option "Modify: create refund, reconcile and create a new draft invoice" then created new invoice not reached from related sale order. I think it we have to fix this issue from sale module, not from account module. I will create new issue and fix from sale module instead of account module. And will remove the change from here.

File: addons/sale/sale.py [hunk 2]
```diff
@@ -824,6 +824,18 @@ def confirm_paid(self):
...
+    @api.model
+    def _refund_cleanup_lines(self, lines):
+        result = super(AccountInvoice, self)._refund_cleanup_lines(lines=lines)
+        for line in lines:
+            values = {}
+            for name, field in line._fields.iteritems():
+                if name == 'sale_line_ids':
+                    values[name] = [(6, 0, line[name].ids)]
+            for r in result:
+                r[2].update(values)
```
Comment (gpa-odoo) [inline addons/sale/sale.py]: This is totally wrong.
It only work for single line. It will broke if it has multiple lines.

Here you assigning `sale_line_ids` of last line to all lines in result
so in result all records will have same `sale_line_ids`

File: addons/sale/sale.py [hunk 3]
```diff
@@ -824,6 +824,18 @@ def confirm_paid(self):
             order.message_post(body=_("Invoice %s paid") % (name))
         return res
 
+    @api.model
+    def _refund_cleanup_lines(self, lines):
+        result = super(AccountInvoice, self)._refund_cleanup_lines(lines=lines)
+        for line in lines:
+            values = {}
+            for name, field in line._fields.iteritems():
```
Comment (gpa-odoo) [inline addons/sale/sale.py]: Why use iteritems() here?  You haven't used `field` in loop

Second thing loop is not required here you can directly write 
`values['sale_line_ids'] = [(6, 0, line['sale_line_ids'].ids)]`
